README.rst

```diff
@@ -460,6 +460,154 @@ template ``context_processors`` in ``settings.py`` as shown below.
         },
     ]
 
+``register_menu_groups``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+It allows to register different menu items or groups at different positions
+in the ``Main navigation menu``
+
+Code example:
+
+.. code-block:: python
+
+    from openwisp_utils.admin_theme.menu import (
+        MenuGroup,
+        MenuItem,
+        MenuLink,
+        register_menu_groups,
+    )
+
+    my_group_items = {
+        1: MenuItem(
+            config={
+                'label': 'Add user',
+                'model': 'auth.User',
+                'name': 'changelist',
+                'icon': 'add-icon',
+            }
+        ),
+        2: MenuLink(
+            config={'label': 'Code', 'url': 'https://openwisp.org/thecode.html'}
+        ),
+    }
+
+    menu_group = {
+        1: MenuGroup(config={'label': 'Docs', 'items': docs_list}),
+        2: MenuItem(
+                config={
+                    'model': 'test_project.Shelf',
+                    'name': 'changelist',
+                    'label': 'View Shelf',
+                    'icon': 'shelf-icon',
+                }
+            )
+        3: MenuLink(config={'label': 'Link Name', 'url': 'https://link.com'})
```

Comment:
I think its good to have class for each item because it help in writing validations, testing and creation of menu much simpler way.
>We parse the data structure and determine what class to use (if the classes are needed at all), this way users only have to pass a data structure and can ignore these 3 classes. It will result simpler to them. It will also result simpler to document.

But it will help user to debug the errors. In many places we are telling user that they have done some mistake in the config of  a menu item or a menu link or a menu group so they don't need to whole code.They can identify mistake quickly. It also creates a good readability in the code. What do you think about this?

Reply:
> help in writing validations, testing and creation of menu much simpler way.

I wonder how, can you please elaborate with an example?

> they don't need inspect the whole code

Why would they need to inspect the whole code in using a json? Why don't we let them know the exact item that failed to parse?

Reply:
>I wonder how, can you please elaborate with an example?

For example, a class object validates itself when they are created. We don't need to parse json to identify the type of menu item. Also, we don't need to parse MENU again before creating its context. we are doing it independently using classes.

>Why would they need to inspect the whole code in using a json? Why don't we let them know the exact item that failed to parse?

We can do this. If you will say then I will change it. Don't you think readability is also important?